pkg/event/controller.go

```diff
@@ -112,7 +117,7 @@ func (gen *Generator) Run(workers int, stopCh <-chan struct{}) {
 	}
 
 	for i := 0; i < workers; i++ {
-		go wait.Until(gen.runWorker, constant.EventControllerResync, stopCh)
+		go wait.Until(gen.runWorker, time.Duration(gen.configHandler.GetBackgroundScanPeriod()), stopCh)
```

Comment:
The `wait.Until` period will simply recreate a new worker each period (if a prior worker exits.) These periods should be set to a small interval e.g. 1s.

In most cases the worker will simply block on the work queue and never exit.